core/jobs/batch_jobs/exp_validation_jobs_test.py

```diff
@@ -0,0 +1,884 @@
+                    self.EXPLORATION_ID_2,
+                    ['state 1']
+                )
+            )
+        ])
+
+    def test_run_with_invalid_state_classifier_id(self) -> None:
+        self.put_multi([self.exp_6, self.opportunity_4])
+        self.assert_job_output_is([
+            job_run_result.JobRunResult.as_stdout('EXPS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stdout('CURATED EXPS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stdout(
+                'INVALID STATE CLASSIFIER SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stderr(
+                'The id of exp is %s and the states having not None '
+                'classifier model id are %s' % (
+                    self.EXPLORATION_ID_6,
+                    [feconf.DEFAULT_INIT_STATE_NAME, 'second state'])
+            )
+        ])
+
+    def test_run_with_invalid_param_changes(self) -> None:
+        self.put_multi([self.exp_7, self.opportunity_5])
+        self.assert_job_output_is([
+            job_run_result.JobRunResult.as_stdout('EXPS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stdout('CURATED EXPS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stdout(
+                'INVALID PARAM CHANGES SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stderr(
+                'The id of exp is %s and the length of '
+                'its param changes is %s' % (
+                    self.EXPLORATION_ID_7, len(self.PARAM_CHANGES_1)
+                )
+            )
+        ])
+
+    def test_run_with_invalid_param_specs(self) -> None:
+        self.put_multi([self.exp_8, self.opportunity_6])
+        self.assert_job_output_is([
+            job_run_result.JobRunResult.as_stdout('EXPS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stdout('CURATED EXPS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stdout(
+                'INVALID PARAM SPECS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stderr(
+                'The id of exp is %s and the length of '
+                'its param specs is %s' % (self.EXPLORATION_ID_8, 2)
+            )
+        ])
+
+    def test_run_with_invalid_training_data(self) -> None:
+        self.put_multi([self.exp_9, self.opportunity_7])
+        self.assert_job_output_is([
+            job_run_result.JobRunResult.as_stdout('EXPS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stdout('CURATED EXPS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stdout(
+                'INVALID TRAINING DATA SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stderr(
+                'The id of exp is %s and the states having interaction '
+                'with non-empty training data are %s' % (
+                    self.EXPLORATION_ID_9,
+                    ['state 1', 'state 2']
+                )
+            )
+        ])
+
+    def test_run_with_invalid_cust_args(self) -> None:
```

Comment:
Can you also test all the other problems separately?

Reply:
The individual tests for all the other problems are already added.